libcodechecker/analyze/plist_parser.py

```diff
@@ -268,3 +271,188 @@ def skip_report_from_plist(plist_file, skip_handler):
+
+    @staticmethod
+    def __format_bug_event(name, severity, event, source_file):
+
+        loc = event['location']
+        fname = os.path.basename(source_file)
+        if name:
+            return '[%s] %s:%d:%d: %s [%s]' % (severity,
+                                               fname,
+                                               loc['line'],
+                                               loc['col'],
+                                               event['message'],
+                                               name)
+        else:
+            return '%s:%d:%d: %s' % (fname,
+                                     loc['line'],
+                                     loc['col'],
+                                     event['message'])
+
+    def __write_bugs(self, output, reports, files, analyzed_source_file,
+                     report_stats):
+        """
+        Write out the bugs to the output and collect report statistics.
+        """
+
+        severity_stats = defaultdict(int)
+        file_stats = defaultdict(int)
+        report_count = defaultdict(int)
+
+        report_num = len(reports)
+        if report_num > 0:
+            index_format = '    %%%dd, ' % \
+                           int(math.floor(math.log10(report_num)) + 1)
+
+        non_suppressed = 0
+        for report in reports:
+            events = [i for i in report.bug_path if i.get('kind') == 'event']
+            f_path = files[events[-1]['location']['file']]
+            if self.skiplist_handler and \
+                    self.skiplist_handler.should_skip(f_path):
+                LOG.debug(report + ' is skipped (in ' + f_path + ")")
+                continue
+            hash_value = report.main['issue_hash_content_of_line_in_context']
+            bug = {'hash_value': hash_value,
+                   'file_path': f_path
+                   }
+            if self.suppress_handler and \
+                    self.suppress_handler.get_suppressed(bug):
+                LOG.debug("Suppressed by suppress file: {0}".format(report))
+                continue
+
+            last_report_event = report.bug_path[-1]
+            source_file = files[last_report_event['location']['file']]
+            report_line = last_report_event['location']['line']
+            report_hash = report.main['issue_hash_content_of_line_in_context']
+            checker_name = report.main['check_name']
+            sp_handler = suppress_handler.SourceSuppressHandler(source_file,
+                                                                report_line,
+                                                                report_hash,
+                                                                checker_name)
+
+            # Check for suppress comment.
+            suppress_data = sp_handler.get_suppressed()
+            if suppress_data:
+                if self.suppress_handler:
+                    LOG.info("Writing source-code suppress at '{0}:{1}' to "
```

Comment:
Is this output true? This should only be printed if `--export-suppress` is given to `parse`. What happens if a suppress file is given but we don't want to export into it?

Reply:
If a suppress file is given than the report will be printed by parse only if it is not suppressed in the suppress file.

Reply:
I understand that. What I meant is the log output here if it holds in execution. I feel this log shouldn't be here.

Reply:
I've changed it to debug log level. It might be useful if there is some problem.